inner-scripts/kubestellar

```diff
@@ -46,6 +46,65 @@ function echoerr() {
    echo "ERROR: $1" >&2
 }
 
+# Check if a given process name is running
+process_running() {
+  SERVICE="$1"
+  if pgrep -f "$SERVICE" >/dev/null
+  then
+      echo "running"
+  else
+      echo "stopped"
+  fi
+}
+
+wait_for_process(){
+  status=$(process_running $1)
+  MAX_RETRIES=10
+  retries=0
+  status_code=0
+  while [ $status != "running" ]; do
+      if [ $retries -eq $MAX_RETRIES ]; then
+           status_code=1
+           break
+      fi
+
+      retries=$(( retries + 1 ))
+      sleep 3
+      status=$(process_running $1)
+  done
+  echo $status_code
+}
+
+function run_kb_konnector_for_imw() {
+    if kubectl get ns kube-system &> /dev/null; then
+        echo "namespace kube-system already exists, skipping"
+    else
+        echo "creating namespace kube-system"
+        kubectl create ns kube-system
+    fi
+    echo "starting kube-bind konnector for imw in background"
+    konnector &> /dev/null &
+
+    run_status=$(wait_for_process konnector)
+    if [ $run_status -eq 0 ]; then
+        echo " konnector started"
+    else
+        echo "konnector not started"
+        sleep 2
```

Comment:
Why is this sleep here?

Reply:
I remember @ezrasilvera used this line to help address some timing issue. But right now I believe this is removed from the `kubestellar` script.